src/plugins/irc-events/link.js

```diff
@@ -166,18 +166,15 @@ function fetch(uri, cb) {
 			}
 		})
 		.on("error", function() {})
-		.pipe(es.map(function(data, next) {
-			length += data.length;
-			if (length > limit) {
-				req.response.req.abort();
+		.on("data", (data) => {
+			if (length + data.length > limit) {
+				req.abort();
+			} else {
+				length += data.length;
+				buffers.push(data);
```

Comment:
You did this differently than in my diff, you're just throwing away last received buffer, remove the `else`.

Reply:
> you're just throwing away last received buffer

What do you mean? It doesn't matter if we push or not the last chunk, since the request gets aborted and the picture not rendered, does it?

Reply:
This code is used to fetch HTML too. You're downloading data and throwing it away for no good reason which is a regression compared to previous code.

Reply:
On that note, we should rename `prefetchMaxImageSize` into `prefetchMaxSize` or something, if it's used for non-images too.

Reply:
It sets the limit to 50kb once it detects a non image.

Reply:
Done.